generator/typesystem.cpp

```diff
@@ -1509,14 +1509,27 @@ TypeDatabase *TypeDatabase::instance()
 
 TypeDatabase::TypeDatabase() : m_suppressWarnings(true)
 {
-    addType(new StringTypeEntry("QString"));
+    StringTypeEntry* mainStringType = new StringTypeEntry("QString");
+    addType(mainStringType);
 
     StringTypeEntry *e = new StringTypeEntry("QLatin1String");
     e->setPreferredConversion(false);
+    e->setEquivalentType(mainStringType);
     addType(e);
 
     e = new StringTypeEntry("QStringRef");
     e->setPreferredConversion(false);
+    e->setEquivalentType(mainStringType);
+    addType(e);
+
+    e = new StringTypeEntry("QStringView");
+    e->setPreferredConversion(false);
+    e->setEquivalentType(mainStringType);
+    addType(e);
+
+    e = new StringTypeEntry("QAnyStringView");
+    e->setPreferredConversion(false);
+    e->setEquivalentType(mainStringType);
     addType(e);
```

Comment:
What about `QUtf8String`? And for that matter, it probably also makes sense to add `QByteArrayView`.

Reply:
You mean QLatin1StringView and QUtf8StringView? I didn't deem them useful, because in most cases there is also a variant of the same method with QAnyStringView or QString as argument.

QByteArrayView might be useful in some cases, but is a little bit more complicated to handle.